src/sentry/incidents/endpoints/project_alert_rule_details.py

```diff
@@ -33,6 +33,7 @@ def put(self, request: Request, project, alert_rule) -> Response:
                 "organization": project.organization,
                 "access": request.access,
                 "user": request.user,
+                "validate_channel_id": True,
```

Comment:
@leeandher recommended that we set the value here to make the endpoint validate the data instead of my fix to raise the errors in another code path.

I believe we should fix the defaults in order that this would never be needed.